include/nbl/asset/IDescriptorSet.h

```diff
@@ -92,31 +93,31 @@ class IDescriptorSet : public virtual core::IReferenceCounted // TODO: try to re
 				}
 				~SDescriptorInfo()
 				{
-					if (desc && desc->getTypeCategory()==IDescriptor::EC_IMAGE)
+					if (desc and (desc->getTypeCategory()==IDescriptor::EC_IMAGE or desc->getTypeCategory() == IDescriptor::EC_SAMPLER))
 						info.image.sampler = nullptr;
 				}
 
 				inline SDescriptorInfo& operator=(const SDescriptorInfo& other)
 				{
-					if (desc && desc->getTypeCategory()==IDescriptor::EC_IMAGE)
+					if (desc and (desc->getTypeCategory()==IDescriptor::EC_IMAGE or desc->getTypeCategory() == IDescriptor::EC_SAMPLER))
 						info.image.sampler = nullptr;
 					desc = other.desc;
 					const auto type = desc->getTypeCategory();
-					if (type!=IDescriptor::EC_IMAGE)
+					if (type!=IDescriptor::EC_IMAGE and type!=IDescriptor::EC_SAMPLER)
```

Comment:
again https://github.com/Devsh-Graphics-Programming/Nabla/pull/693/files#r1648875709

so you need a `switch (type)` probably:
```
case EC_BUFFER:
   info.buffer = other.info.buffer;
   break;
case EC_IMAGE:
   info.image= other.info.image;
   break;
default: // the rest don't have layouts, extra samplers, or buffer ranges (even buffer view
   break;
```